src/consensus/role_timeout.rs

```diff
@@ -163,27 +241,70 @@ impl TimeoutRole for Consensus {
+            Signed {
+                msg: ConsensusNetMessage::Timeout(Signed { msg: (s, v, p), .. }, ..),
+                ..
+            } => (s, v, p),
+            _ => {
+                bail!("Received timeout message with unexpected format");
+            }
+        };
+
+        if received_parent_hash != &self.bft_round_state.parent_hash {
+            debug!(
+                "🌘 Ignoring timeout with incorrect parent hash {}, expected {}",
+                received_parent_hash, self.bft_round_state.parent_hash
+            );
+            return Ok(());
+        }
+        if received_slot < &self.bft_round_state.slot {
             debug!(
                 "🌘 Ignoring timeout for slot {}, am at {}",
                 received_slot, self.bft_round_state.slot
             );
             return Ok(());
         }
 
-        if received_slot != self.bft_round_state.slot || received_view != self.bft_round_state.view
+        if received_slot != &self.bft_round_state.slot
+            || received_view != &self.bft_round_state.view
         {
-            bail!(
+            info!(
                 "Timeout (Slot: {}, view: {}) does not match expected (Slot: {}, view: {})",
-                received_slot,
-                received_view,
-                self.bft_round_state.slot,
-                self.bft_round_state.view,
+                received_slot, received_view, self.bft_round_state.slot, self.bft_round_state.view,
             );
+            return Ok(());
         }
 
-        // In the paper, a replica returns a commit if present
-        // TODO ?
+        // If there is a prepareQC along with this message, verify it (we can, it's the same slot),
+        // and then potentially update our highest seen PrepareQC.
+        if let Signed {
+            msg: ConsensusNetMessage::Timeout(_, TimeoutKind::PrepareQC((qc, cp))),
+            ..
+        } = &received_msg
+        {
+            if cp.slot == *received_slot {
+                self.verify_quorum_certificate(ConsensusNetMessage::PrepareVote(cp.hashed()), qc)
+                    .context("Verifying PrepareQC")?;
+                if self
+                    .store
+                    .bft_round_state
+                    .timeout
+                    .update_highest_seen_prepare_qc(*received_slot, *received_view, qc.clone())
+                {
+                    // Update our consensus proposal
+                    self.bft_round_state.current_proposal = cp.clone();
+                    debug!("Highest seen PrepareQC updated");
+                }
+            } else {
+                // We actually cannot process this, or we might end up thinking we have 2f+1 timeouts but not working.
+                bail!(
+                    "Received timeout message with prepareQC for slot {}, but we are at {}",
+                    cp.slot,
+                    received_slot
+                );
```

Comment:
J'ai peut-être mal compris le code, mais le message me semble pas bon
J'aurais plutot mis 
```suggestion
                bail!(
                    "Received incorrect timeout message. PrepareQC is for slot {}, but Timeout is about slot {}",
                    cp.slot,
                    received_slot
                );
```